Source/WebKit/WPEPlatform/wpe/WPEApplicationAccessibleAtk.cpp

```diff
@@ -0,0 +1,136 @@
+#include <wtf/Vector.h>
+#include <wtf/glib/GUniquePtr.h>
+#include <wtf/glib/WTFGType.h>
+
+struct _WPEApplicationAccessibleAtkPrivate {
+    Vector<WPEToplevel*> toplevels;
+};
+
+WEBKIT_DEFINE_FINAL_TYPE(WPEApplicationAccessibleAtk, wpe_application_accessible_atk, ATK_TYPE_OBJECT, AtkObject)
+
+static void wpeApplicationAccessibleAtkConstructed(GObject* object)
+{
+    G_OBJECT_CLASS(wpe_application_accessible_atk_parent_class)->constructed(object);
+
+    auto* accessible = WPE_APPLICATION_ACCESSIBLE_ATK(object);
+    GUniquePtr<GList> toplevels(wpeToplevelList());
+    for (GList* iter = toplevels.get(); iter; iter = g_list_next(iter))
+        accessible->priv->toplevels.append(WPE_TOPLEVEL(iter->data));
+}
+
+static void wpeApplicationAccessibleAtkInitialize(AtkObject* atkObject, gpointer userData)
+{
+    ATK_OBJECT_CLASS(wpe_application_accessible_atk_parent_class)->initialize(atkObject, userData);
+
+    atkObject->role = ATK_ROLE_APPLICATION;
+    atkObject->accessible_parent = nullptr;
+}
+
+static gint wpeApplicationAccessibleAtkGetNChildren(AtkObject* atkObject)
+{
+    auto* accessible = WPE_APPLICATION_ACCESSIBLE_ATK(atkObject);
+    return accessible->priv->toplevels.size();
+}
+
+static AtkObject* wpeApplicationAccessibleAtkRefChild(AtkObject* atkObject, int index)
+{
+    auto* accessible = WPE_APPLICATION_ACCESSIBLE_ATK(atkObject);
+    if (index < 0 || static_cast<size_t>(index) >= accessible->priv->toplevels.size())
+        return nullptr;
+
+    auto* toplevel = accessible->priv->toplevels[index];
+    auto* toplevelAccessible = wpeToplevelGetOrCreateAccessibleAtk(toplevel);
+    if (!ATK_IS_OBJECT(toplevelAccessible))
+        return nullptr;
+
+    return ATK_OBJECT(g_object_ref(toplevelAccessible));
+}
+
+static const char* wpeApplicationAccessibleAtkGetName(AtkObject*)
+{
+    return g_get_prgname();
+}
+
+static void wpe_application_accessible_atk_class_init(WPEApplicationAccessibleAtkClass* klass)
+{
+    GObjectClass* objectClass = G_OBJECT_CLASS(klass);
+    objectClass->constructed = wpeApplicationAccessibleAtkConstructed;
+
+    AtkObjectClass* atkObjectClass = ATK_OBJECT_CLASS(klass);
+    atkObjectClass->initialize = wpeApplicationAccessibleAtkInitialize;
+    atkObjectClass->get_n_children = wpeApplicationAccessibleAtkGetNChildren;
+    atkObjectClass->ref_child = wpeApplicationAccessibleAtkRefChild;
+    atkObjectClass->get_name = wpeApplicationAccessibleAtkGetName;
+    atkObjectClass->get_parent = nullptr;
+}
+
+AtkObject* wpeApplicationAccessibleAtkNew()
+{
+    auto* accessible = ATK_OBJECT(g_object_new(WPE_TYPE_APPLICATION_ACCESSIBLE_ATK, nullptr));
+    atk_object_initialize(accessible, nullptr);
+    return accessible;
+}
+
+void wpeApplicationAccessibleAtkToplevelCreated(WPEApplicationAccessibleAtk* accessible, WPEToplevel* toplevel)
+{
+    auto* toplevelAccessible = wpeToplevelGetOrCreateAccessibleAtk(toplevel);
+    if (!ATK_IS_OBJECT(toplevelAccessible))
+        return;
```

Comment:
Same comment as above... Shouldn't creation of an accessible for toplevels always succed?